File: src/parser/cpp/source/parsetools.cpp
#include "./structs.hpp"

#include <string>

using namespace std;

string tkstr(StateParse &S, const int &tid) {
	if (tid == -1) return "";
	// Return interpolated string for string tokens.
	Token &tk = S.LexerData.tokens[tid];
	if (tk.kind == "tkSTR") {
		if (!tk.$.empty()) return tk.$;
	}
	int start = tk.start;
	int end = tk.end;
	return S.text.substr(start, end - start + 1);
}

File: src/parser/go/main/utils/parsetools/parsetools.go
package parsetools

import (
	"github.com/cgabriel5/compiler/utils/structs"
)

type StateParse = structs.StateParse

func Tkstr(S *StateParse, tid int) string {
	if tid == -1 {
		return ""
	}
	// Return interpolated string for string tokens.
	tk := &(S.LexerData.Tokens[tid])
	if tk.Kind == "tkSTR" {
		if tk.Str_rep != "" {
			return tk.Str_rep
		}
	}
	start := tk.Start
	end := tk.End
	return S.Text[start : end+1]
}
